sdk/search/azure-search/src/main/java/com/azure/search/documents/SearchApiKeyPipelinePolicy.java

```diff
@@ -12,31 +13,37 @@
 import java.util.Objects;
 
 /**
- * Policy that adds the Cognitive Search Service api-key into the request's Authorization header.
+ * Pipeline policy that uses an {@link AzureKeyCredential} to set the {@code api-key} header which authorizes requests
+ * sent to the Azure Search service.
+ *
+ * <p>
+ * Requests sent with this pipeline policy are required to use {@code HTTPS}. If the request isn't using {@code HTTPS}
+ * an exception will be thrown to prevent leaking the key.
  */
 public final class SearchApiKeyPipelinePolicy implements HttpPipelinePolicy {
```

Comment:
Could the policies be private?

Reply:
Java has a guideline where we're expected to have our pipeline policies in the public API. We allow customers to roll their own pipeline if they wish, so making these inaccessible from the public API would make that more difficult.

@JonathanGiles to point to the guideline since I'm having trouble finding it. (Or to add it if it is missing)

Reply:
Thanks for the clarification Alan!

Reply:
The wording might be unclear but it's [here](https://azure.github.io/azure-sdk/java_implementation.html#java-auth-policy-impl) - "DO provide a suitable authentication policy that authenticates the HTTP request in the HTTP pipeline when using non-standard credentials. This includes custom connection strings, if supported."

Reply:
In this case I would push to move up the shared Azure.Core implementation of a generic auth policy as in the discussion below, so we don't have these types left around as artifacts in libraries that GA before it's available

Reply:
If it is okay, I'm going to merge this PR as it keeps hitting merge conflicts due to its scope and file an issue to follow-up on a generic policy.